dotnet/src/Connectors/Connectors.Memory.AzureAISearch/AzureAISearchMemoryRecordService.cs

```diff
@@ -0,0 +1,357 @@
+    }
+
+    /// <summary>
+    /// Convert the public <see cref="GetRecordOptions"/> options model to the azure ai search <see cref="GetDocumentOptions"/> options model.
+    /// </summary>
+    /// <param name="options">The public options model.</param>
+    /// <returns>The azure ai search options model.</returns>
+    private GetDocumentOptions ConvertGetDocumentOptions(GetRecordOptions? options)
+    {
+        var innerOptions = new GetDocumentOptions();
+        if (options?.IncludeVectors is false)
+        {
+            innerOptions.SelectedFields.AddRange(this._nonVectorPropertyNames);
+        }
+
+        return innerOptions;
+    }
+
+    /// <summary>
+    /// Run the given operation and wrap any <see cref="RequestFailedException"/> with <see cref="HttpOperationException"/>."/>
+    /// </summary>
+    /// <typeparam name="T">The response type of the operation.</typeparam>
+    /// <param name="operation">The operation to run.</param>
+    /// <param name="collectionName">The name of the collection the operation is being run on.</param>
+    /// <param name="operationName">The type of database operation being run.</param>
+    /// <returns>The result of the operation.</returns>
+    private static async Task<T> RunOperationAsync<T>(Func<Task<T>> operation, string collectionName, string operationName)
+    {
+        try
+        {
+            return await operation.Invoke().ConfigureAwait(false);
+        }
+        catch (RequestFailedException ex)
+        {
+            var wrapperException = new MemoryServiceOperationException("Call to memory service failed.", ex);
+
+            wrapperException.Data.Add("db.system", "AzureAISearch");
+            wrapperException.Data.Add("db.collection.name", collectionName);
+            wrapperException.Data.Add("db.operation.name", operationName);
+
+            throw wrapperException;
+        }
+    }
+
+    /// <summary>
+    /// Run the given model conversion and wrap any exceptions with <see cref="MemoryModelException"/>.
+    /// </summary>
+    /// <typeparam name="T">The response type of the operation.</typeparam>
+    /// <param name="operation">The operation to run.</param>
+    /// <param name="collectionName">The name of the collection the operation is being run on.</param>
+    /// <param name="operationName">The type of database operation being run.</param>
+    /// <returns>The result of the operation.</returns>
+    private static T RunModelConversion<T>(Func<T> operation, string collectionName, string operationName)
+    {
+        try
+        {
+            return operation.Invoke();
+        }
+        catch (Exception ex)
+        {
+            var wrapperException = new MemoryModelException("Failed to convert memory model.", ex);
+
+            wrapperException.Data.Add("db.system", "AzureAISearch");
+            wrapperException.Data.Add("db.collection.name", collectionName);
+            wrapperException.Data.Add("db.operation.name", operationName);
```

Comment:
It's using the open telemetry specification, see: https://opentelemetry.io/docs/specs/semconv/attributes-registry/db/
Mark ended up using the http fields names from the same spec for http failures. Since we have to use some names, we figured it may as well come from a standard.  I'll add a comment to clarify this, because unless it's obvious where this comes from, on extension someone might accidentally add values from outside the spec.